spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/handler/predicate/QueryRoutePredicateFactory.java

```diff
@@ -110,6 +121,26 @@ public Config setRegexp(String regexp) {
 			return this;
 		}
 
+		public Predicate<String> getPredicate() {
+			return predicate;
+		}
+
+		public Config setPredicate(Predicate<String> predicate) {
+			this.predicate = predicate;
+			return this;
+		}
+
+		/**
+		 * Enforces the validation done on predicate configuration: {@link #regexp} and
+		 * {@link #predicate} can't be both set at runtime.
+		 * @return <code>false</code> if {@link #regexp} and {@link #predicate} are both
+		 * set in this predicate factory configuration
+		 */
+		@AssertTrue
```

Comment:
Remove this

Reply:
Hi, I added the `isValid` method and its `@AssertTrue` annotation from the validation framework to ensure that at most one between `regexp` or `predicate` is specified, and to let this be checked at the same time of the other constraints. I don't understand if, removing this, there is still a way to check this condition during the validation phase or not. Is the simple `isValid` method enough to ensure this?

Reply:
ah, you can leave it.